frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -165,7 +166,11 @@ export async function request(
   const controller = new AbortController();
   const id = setTimeout(() => controller.abort(), timeout);
 
-  const url = combinePath(BASE_HTTP_URL, fullPath);
+  let url = combinePath(BASE_HTTP_URL, fullPath);
+  if (queryParams) {
+    const query = new URLSearchParams(queryParams);
+    url = `${url}?${query.toString()}`;
+  }
```

Comment:
We have as of very recently this function called `asQuery` which does most of this.